app/models/webhook.rb

```diff
@@ -1,11 +1,19 @@
 class Webhook < ActiveRecord::Base
   has_soft_deletion default_scope: true
-  validates :branch, uniqueness: { scope: [ :stage], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
+  validates :branch, uniqueness: { scope: [ :stage ], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
 
   belongs_to :project
   belongs_to :stage
 
+  SOURCES = Rails.root.join('app','controllers', 'integrations').children(false).map do |controller_path|
```

Comment:
What if this class `require`s everything in `controlllers/integrations` and then uses the `IntegrationsController.descendants` method?